art/metrics/metrics.py

```diff
@@ -73,12 +78,53 @@ def get_crafter(classifier: "CLASSIFIER_TYPE", attack: str, params: Optional[Dic
     except Exception:  # pragma: no cover
         raise NotImplementedError(f"{attack} crafting method not supported.") from Exception
 
+    if "params" in SUPPORTED_METHODS[attack]:
+        crafter.set_params(**SUPPORTED_METHODS[attack]["params"])
     if params:
         crafter.set_params(**params)
 
     return crafter
 
 
+def adversarial_accuracy(
+    classifier: "CLASSIFIER_TYPE",
+    x: np.ndarray,
+    attack_name: str = None,
+    attack_params: Optional[Dict[str, Any]] = None,
+    attack_crafter: EvasionAttack = None,
+) -> float:
+    """
+    Compute the adversarial accuracy of a classifier object over the sample `x` for a given adversarial crafting
+    method `attack`. `attack_name` can be specified to use a preset attack, or `attack_crafter` for wider choices
+    and customized parameters. Note that the score does not exclude wrong predictions by the classifier.
+
+    :param classifier: A trained model.
+    :param x: Input samples of shape that can be fed into `classifier`.
+    :param attack_name: A string specifying the attack to be used. Currently supported attacks are {`fgsm', `hsj`}
+                        (Fast Gradient Sign Method, Hop Skip Jump).
```

Comment:
I think we need to update the available attacks.

Reply:
I changed the description to, basically refer to SUPPORTED_METHODS as it can evolve. I also modified the code to print available attack names when the input was wrong.